ledger/heavy/executor/backuper.go

```diff
@@ -0,0 +1,293 @@
+///
+// Copyright 2019 Insolar Technologies GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+///
+
+package executor
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/insolar/insolar/configuration"
+	"github.com/insolar/insolar/insolar"
+	"github.com/insolar/insolar/insolar/store"
+	"github.com/insolar/insolar/instrumentation/inslogger"
+	"github.com/pkg/errors"
+)
+
+// BackupMaker is interface for doing backups
+type BackupMaker interface {
+	// Do starts process of incremental backups
+	Do(ctx context.Context, lastFinalizedPulse insolar.PulseNumber) error
```

Comment:
`DoIncremental` or `DoSince` or `BackupSince` or just `Backup`

Reply:
or `MakeBackup` (if it's "maker")

Reply:
Why just 'Do' is bad?